### ISSUE TITLE
coverage compilation fails for scala 3 enum extending java Enum

### ISSUE THREAD

**Issue by rolman243** [scoverage/sbt-scoverage#464]
scoverage version: 2.0.4
scala versoin: 3.2.0
sbt version: 1.7.1

The following code fails to compile with `sbt clean coverage test` but compiles fine with `sbt clean compile test`
```
import scribe.Level

enum MetricLogLevel(val level: Level) extends Enum[MetricLogLevel]:
  case Warn  extends MetricLogLevel(Level.Warn)
  case Error extends MetricLogLevel(Level.Error)
  case Fatal extends MetricLogLevel(Level.Fatal)
```

Compiler failure:
```
[error]   |enum MetricLogLevel(val level: Level) extends Enum[MetricLogLevel]:
[error]   |                                              ^^^^^^^^^^^^^^^^^^^^
[error]   |wrong number of arguments at constructors for (_$name: String, _$ordinal: Int): Enum: (MetricLogLevel.super.<init> : (_$name: String, _$ordinal: Int): Enum), expected: 2, found: 0

[error]   |  case Warn  extends MetricLogLevel(Level.Warn)
[error]   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]   |wrong number of arguments at constructors for (level: scribe.Level, _$name: String, _$ordinal: Int): 
[error]   |  fdn.core.log.MetricLogLevel: ($anon.super.<init> : 
[error]   |  (level: scribe.Level, _$name: String, _$ordinal: Int): 
[error]   |    fdn.core.log.MetricLogLevel
[error]   |), expected: 3, found: 1

[error]   |  case Error extends MetricLogLevel(Level.Error)
[error]   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]   |wrong number of arguments at constructors for (level: scribe.Level, _$name: String, _$ordinal: Int): 
[error]   |  fdn.core.log.MetricLogLevel: ($anon.super.<init> : 
[error]   |  (level: scribe.Level, _$name: String, _$ordinal: Int): 
[error]   |    fdn.core.log.MetricLogLevel
[error]   |), expected: 3, found: 1

[error]   |  case Fatal extends MetricLogLevel(Level.Fatal)
[error]   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]   |wrong number of arguments at constructors for (level: scribe.Level, _$name: String, _$ordinal: Int): 
[error]   |  fdn.core.log.MetricLogLevel: ($anon.super.<init> : 
[error]   |  (level: scribe.Level, _$name: String, _$ordinal: Int): 
[error]   |    fdn.core.log.MetricLogLevel
[error]   |), expected: 3, found: 1
```

The documentation for constructor of the abstract java Enum class states:
```
    /**
     * Sole constructor.  Programmers cannot invoke this constructor.
     * It is for use by code emitted by the compiler in response to
     * enum class declarations.
     *
     * @param name - The name of this enum constant, which is the identifier
     *               used to declare it.
     * @param ordinal - The ordinal of this enumeration constant (its position
     *         in the enum declaration, where the initial constant is assigned
     *         an ordinal of zero).
     */
    protected Enum(String name, int ordinal) {
        this.name = name;
        this.ordinal = ordinal;
    }
```

**Comment by ckipp01** [comment]
Hey @rolman243 thanks for the report. sbt-scoverage is just a thin wrapper facilitation everything. For Scala 3 the coverage logic is actually included in the compiler. Could you report this [there](https://github.com/lampepfl/dotty/issues)?

**Comment by rolman243** [comment]
@ckipp01 yes of course, I'll open an issue

**Comment by TheElectronWill** [comment]
thanks for the issue!

**Comment by TheElectronWill** [comment]
this is fixed in dotty >= 3.2.2-RC1 :)

**Comment by rolman243** [comment]
works with dotty 3.2.2
